### ISSUE TITLE
BuildMemberRef signature always returns void and always takes argument string

### ISSUE THREAD

**Issue by ViRb3** [ioncodes/dnpatch#17]
Isn't this going to fail to match the original members with a different signature?

**Comment by ioncodes** [comment]
Example?

**Comment by ViRb3** [comment]
Say I want to find a call to `System.Console Read`.
I create an instruction to find like in your example project:
```csharp
Instruction.Create(OpCodes.Call, p.BuildMemberRef("System", "Console", "Read", Patcher.MemberRefType.Static)),
```

`BuildMemberRef()` does this:
```csharp
return new MemberRefUser(_module, "Read",
                    MethodSig.CreateStatic(_module.CorLibTypes.Void, _module.CorLibTypes.String),
                    typeRef);
```

So it creates a MemberRef with signature like this:
```
Parameters: String
Returns: Void
````

But the real signature is:
```
Parameters: None
Returns: String
````

Thus, I'm pretty confident it will fail in finding it. I don't have access to a VS environment so I haven't tested anything, I may be wrong.

**Comment by ViRb3** [comment]
E.g. here: https://github.com/ioncodes/dnpatch/blob/master/dnpatch/PatchHelper.cs#L186

**Comment by ioncodes** [comment]
Thanks for the report, I will look into it ASAP!

**Comment by SlowLogicBoy** [comment]
this should be fixed with #13 since `BuildMemberRef` is supported only for `Console.WriteLine()` and similar signature methods

**Comment by ioncodes** [comment]
https://github.com/ioncodes/dnpatch/commit/efd3dbd83237e2c8229db25c60e12654a69453f1
